datasets/dataforgood-fb-hrsl.yaml

```diff
@@ -0,0 +1,36 @@
+Name: High Resolution Population Density Maps + Demographic Estimates by CIESIN and Facebook
+Description: |
+  The population of the world, allocated to 1 arcsecond blocks. This
+  refines CIESIN’s Gridded Population of the World project, using
+  machine learning models on high-resolution worldwide Digital Globe
+  satellite imagery.
+Documentation: https://dataforgood.fb.com/approach/population-density-maps-documentation/
```

Comment:
We needed to update the documentation URL, would you mind editing it to https://dataforgood.fb.com/population-density-maps-documentation/ prior to merge?